test_elasticsearch_dsl/test_update_by_query.py

```diff
@@ -0,0 +1,286 @@
+from copy import deepcopy
+
+from elasticsearch_dsl import UpdateByQuery, query, Q, Document
+
+
+# A lot of the basic query testing for Search can be copied over here and re-used
```

Comment:
there is no point in copying unit tests when your class uses the parent methods, it just introduces duplications. If you are not reimplmenting methods like `index()` do not copy their tests here.

Reply:
Thanks for your comments - I'm going to start working on them today.

But one question here: the UpdateByQuery doesn't inherit from Search because the query capabilities are so different - I can slim down the tests used here. Or would you rather I refactor to use UpdateByQuery inherits from Search to reduce duplication?

Reply:
True, but some of the tests here (like `.index()`) test the functionality of `Request` that `UpdateByQuery` does extend. Also it might be ok to create another class below `Request` that both `Search` and `UpdateByQuery` can then extend that would contain the common functionality, what do you think?

Reply:
Looking at it further it's just `query`, `filter`, and `exclude`. I would be ok with just a simple test each to verify they work since it's the same code as in `Search` where it is tested more extensively. Alternatively another class (either to extend or to mix in) is always a solution too.